### PR TITLE
[v7r2] fix (DMS): directory-sync: better failure logging, later abort

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES

*DMS
FIX: dirac-dms-directory-sync: The script will now try all files and not end a thread if one of the files assigned to be uploaded or downloaded to a thread fails. The script will also print out all failed files, and their error at the end of a thread. A few log messages have been corrected as well.

ENDRELEASENOTES

- [x] Draft PR: pending testing

### CODE REVIEW

File: src/DIRAC/DataManagementSystem/scripts/dirac_dms_directory_sync.py [hunk 1]
```diff
@@ -232,21 +220,21 @@ def uploadLocalFile(dm, lfn, localfile, storage):
         """
         Upload a local file to a storage element
         """
-        res = dm.putAndRegister(lfn, localfile, storage, None)
+        res = returnSingleResult(dm.putAndRegister(lfn, localfile, storage, None))
```
Comment (chaen) [inline src/DIRAC/DataManagementSystem/scripts/dirac_dms_directory_sync.py:223]: can't that whole `uploadLocalFile` be replaced with `returnSingleResult(dm.putAndRegister(lfn, localfile, storage, None))` ?

Comment (andresailer) [reply src/DIRAC/DataManagementSystem/scripts/dirac_dms_directory_sync.py:223]: Yes, it can be replaced like that

Comment (andresailer) [reply src/DIRAC/DataManagementSystem/scripts/dirac_dms_directory_sync.py]: Replaced this function, and downloadFile function

**Comment by andresailer** [comment]
PR is ready for review

**Comment by andresailer** [comment]
Rebased, maybe that fixes the doc test failure...

**Comment by andresailer** [comment]
Documentation now successful. Only the rucio agents tests are failing now  (#5637 )

**Review by atsareg — APPROVED** [review]

**Pull request merged**